File: 图片轮播器－oc/图片轮播器－oc/ViewController.m
//
//  ViewController.m
//  图片轮播器－oc
//
//  Created by zz on 16/2/27.
//  Copyright © 2016年 zz. All rights reserved.
//

#import "ViewController.h"
#import "ZDCollectionView.h"

@interface ViewController ()
@property (strong, nonatomic) NSArray *urls;

@end

@implementation ViewController

-(NSArray *)urls{

    if (_urls == nil) {
        
        NSMutableArray *temp = [NSMutableArray array];
        for (int i = 0; i< 3; i++) {
            
            NSString *path = [[NSBundle mainBundle]pathForResource:[NSString stringWithFormat:@"%02d",i+1] ofType:@"jpg"];
            NSURL *url = [NSURL fileURLWithPath:path];
            [temp addObject:url];
        }
        _urls = temp.copy;
    }
    return _urls;
}
- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.

    ZDCollectionView *pictureCollectionView = [[ZDCollectionView alloc]initWithURLs:self.urls didSelected:nil];
    
    pictureCollectionView.frame = CGRectMake(20, 20, self.view.bounds.size.width - 40, 200);
    
    [self.view addSubview:pictureCollectionView];

}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: 图片轮播器swift/图片轮播器swift/ViewController.swift
//
//  ViewController.swift
//  图片轮播器－swift
//
//  Created by zz on 16/2/28.
//  Copyright © 2016年 zz. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    var urls: [NSURL] = [NSURL]()
    

    
    override func viewDidLoad() {
        super.viewDidLoad()
        loadUrl()
        let loopView: LoopView = LoopView(urls: urls)

        loopView.frame = CGRect(x: 10, y: 40, width: 300, height: 200)
        view.addSubview(loopView)
        
        loopView.callBack = {selected in print("选中了\(selected)")}
    }

    private func loadUrl(){
    
        for i in 1..<4 {
          
            let path = NSBundle.mainBundle().pathForResource("0\(i)", ofType: "jpg")
            let url = NSURL(fileURLWithPath: path!)
            
            urls.append(url)
            
        }
        
    }


}


